lib/functional.js

```diff
@@ -19,4 +19,89 @@ module.exports = (api) => {
     value !== undefined && value !== null ? fn(value) : defVal
   );
 
+  // Either can be used for saving either left type or right type value.
+  // Left type is commonly used for additional information like error.
+  //
+  function Either(
+    // Either can be used for saving either left type or right type value.
+    // Left type is commonly used for additional information like error.
+    left, // left type value
+    right // right type value
+    // Returns tuple containing left type value or null as first element
+    // and right type value or undefined as second element
+  ) {
+    if (left) {
+      this.isLeft = true;
+      this.val = left;
+    } else {
+      this.isLeft = false;
+      this.val = right;
+    }
+  }
+
+  api.common.Either = Either;
+
+  api.common.eitherRight = (
+    // Function for creating either values with right type inside
+    right // right type value
+    // Returns either value with right type inside
+  ) => (new Either(null, right));
+
+  api.common.altAppend = (
+    // Alternative append for either values
+    e1, // first either value
+    e2 // second either value
+    // Returns result of appending two either values
+  ) => (e1[0] ? e1 : e2);
+
+  api.common.altConcat = (
+    // Alternative concatenation for either values
+    ...args // arguments for concatenation
+    // Returns result of concatenation of args
+  ) => {
+    switch (args.length) {
```

Comment:
I'd prefer not using `switch` here, better `if (args.length === 0) return new Error('...');`